legion/engine/llri-vk/detail/device.cpp

```diff
@@ -239,41 +241,94 @@ namespace llri
-        auto r = static_cast<VolkDeviceTable*>(m_functionTable)->vkAllocateMemory(static_cast<VkDevice>(m_ptr), &allocInfo, nullptr, &memory);
+        auto r = table->vkAllocateMemory(static_cast<VkDevice>(m_ptr), &allocInfo, nullptr, &memory);
         if (r != VK_SUCCESS)
         {
             if (isTexture)
-                static_cast<VolkDeviceTable*>(m_functionTable)->vkDestroyImage(static_cast<VkDevice>(m_ptr), image, nullptr);
+                table->vkDestroyImage(static_cast<VkDevice>(m_ptr), image, nullptr);
             else
-                static_cast<VolkDeviceTable*>(m_functionTable)->vkDestroyBuffer(static_cast<VkDevice>(m_ptr), buffer, nullptr);
+                table->vkDestroyBuffer(static_cast<VkDevice>(m_ptr), buffer, nullptr);
             return internal::mapVkResult(r);
         }
 
         if (isTexture)
-            r = static_cast<VolkDeviceTable*>(m_functionTable)->vkBindImageMemory(static_cast<VkDevice>(m_ptr), image, memory, 0);
+            r = table->vkBindImageMemory(static_cast<VkDevice>(m_ptr), image, memory, 0);
         else
-            r = static_cast<VolkDeviceTable*>(m_functionTable)->vkBindBufferMemory(static_cast<VkDevice>(m_ptr), buffer, memory, 0);
+            r = table->vkBindBufferMemory(static_cast<VkDevice>(m_ptr), buffer, memory, 0);
 
         if (r != VK_SUCCESS)
         {
             if (isTexture)
-                static_cast<VolkDeviceTable*>(m_functionTable)->vkDestroyImage(static_cast<VkDevice>(m_ptr), image, nullptr);
+                table->vkDestroyImage(static_cast<VkDevice>(m_ptr), image, nullptr);
             else
-                static_cast<VolkDeviceTable*>(m_functionTable)->vkDestroyBuffer(static_cast<VkDevice>(m_ptr), buffer, nullptr);
+                table->vkDestroyBuffer(static_cast<VkDevice>(m_ptr), buffer, nullptr);
 
-            static_cast<VolkDeviceTable*>(m_functionTable)->vkFreeMemory(static_cast<VkDevice>(m_ptr), memory, nullptr);
+            table->vkFreeMemory(static_cast<VkDevice>(m_ptr), memory, nullptr);
 
             return internal::mapVkResult(r);
         }
+        
+        // this part is necessary because in vulkan images are created int he UNDEFINED layout
+        // so we must transition them to desc.initialState manually.
+        if (isTexture)
+        {
+            table->vkResetCommandPool(static_cast<VkDevice>(m_ptr), static_cast<VkCommandPool>(m_workCmdGroup), {});
+            
+            // Record commandbuffer to transition texture from undefined
+            VkCommandBufferBeginInfo beginInfo {};
+            beginInfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO;
+            beginInfo.pNext = nullptr;
+            beginInfo.flags = VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT;
+            beginInfo.pInheritanceInfo = nullptr;
+            table->vkBeginCommandBuffer(static_cast<VkCommandBuffer>(m_workCmdList), &beginInfo);
+            
+            // use the texture format to detect the aspect flags
```

Comment:
You're a little inconsistent in whether you start comments with upper case letters or lower case. whether that's an issue i'll leave up to you

Reply:
I suppose that's a bit of a concern, I'm not sure what my preference is for comments

Reply:
I think I'm going to leave this to a future PR where I'll establish a standard for API & implementation comments and make it consistent throughout